### DISCUSSION TITLE
Tasmota console message that disables the serial port after 10 minutes

### DISCUSSION THREAD

**Discussion by cesarmorisco** [arendst/Tasmota#15343] [Q&A] [answered]
Hello, I have a doubt about Tasmota because it disables the serial port in 10 minutes. And how to leave it active to communicate with a microcontroller. Thank you

**Comment by sfromis** [comment, marked as answer]
Tasmota does not disable the serial port after 10 minutes. If you use it for serial console, it will only cease producing output. It is still available for communication, like issuing commands. To avoid it ceasing logging, issue the SerialLog command to set log level (changed or not). You can do that from a boot time rule if you want it every boot.

If your usage is not serial console, there should be no issue.

**Reply by barbudor** [reply]
@cesarmorisco I already have provided answers to you in the other topic which I copy here for references:
> @cesarmorisco by design Tasmota disable the serial console after 10 minutes
Normally you wouldn't use the Serial console to talk with a MCU but [Serial Bridge](https://tasmota.github.io/docs/Commands/#serial-bridge) or [TasmotaClient](https://tasmota.github.io/docs/TasmotaClient/)
Now, if you insist in keeping the serial console active you can have a rule such as ON Time#minute|5 DO SerialLog 2 ENDON

Have you read the links I provided ?
Why don't you ask the proper question ?

**Reply by ascillato2** [reply]
Tasmota disables the serial log after 10 minutes. Not the serial port. The serial port continues working. If you send or receive any command it will work. 

Please, take a look to the Tasmota documentation. Thanks.

**Comment by cesarmorisco** [comment]
![Tasmota2](https://user-images.githubusercontent.com/19960084/162786529-26f2bc95-36af-4f7a-ab28-26ef7fde9326.png)
Thanks for the answer. Here the settings I'm using. Here the serial report is shutting down.



00:00:00 CFG: Lido da flash em FA, Contagem 66
00:00:00 Projeto tasmota Tasmota Versão 7.2.0(tasmota)-2_6_1
00:00:00 WIF: Ligando ao PA1 Help-Info Central Tec 2.4G em modo 11N como tasmota-6654...
00:00:06 WIF: Ligado
00:00:06 HTP: Servidor WEB ativo em tasmota-6654 com o endereço IP 10.0.0.29
00:00:13 RSL: stat/tasmota/RESULT = {"POWER2":"OFF"}
00:00:13 RSL: stat/tasmota/POWER2 = OFF
00:00:14 RSL: stat/tasmota/RESULT = {"POWER1":"OFF"}
00:00:14 RSL: stat/tasmota/POWER1 = OFF
00:05:06 RSL: tele/tasmota/STATE = {"Time":"1970-01-01T00:05:06","Uptime":"0T00:05:08","UptimeSec":308,"Heap":28,"SleepMode":"Dynamic","Sleep":50,"LoadAvg":19,"MqttCount":0,"POWER1":"OFF","POWER2":"OFF","Wifi":{"AP":1,"SSId":"Help-Info Central Tec 2.4G","BSSId":"24:FD:0D:69:B7:2D","Channel":11,"RSSI":74,"Signal":-63,"LinkCount":1,"Downtime":"0T00:00:07"}}
00:09:58 APP: Registro Serial desabilitado
00:10:06 RSL: tele/tasmota/STATE = {"Time":"1970-01-01T00:10:06","Uptime":"0T00:10:08","UptimeSec":608,"Heap":28,"SleepMode":"Dynamic","Sleep":50,"LoadAvg":19,"MqttCount":0,"POWER1":"OFF","POWER2":"OFF","Wifi":{"AP":1,"SSId":"Help-Info Central Tec 2.4G","BSSId":"24:FD:0D:69:B7:2D","Channel":11,"RSSI":64,"Signal":-68,"LinkCount":1,"Downtime":"0T00:00:07"}}
00:15:06 RSL: tele/tasmota/STATE = {"Time":"1970-01-01T00:15:06","Uptime":"0T00:15:08","UptimeSec":908,"Heap":28,"SleepMode":"Dynamic","Sleep":50,"LoadAvg":19,"MqttCount":0,"POWER1":"OFF","POWER2":"OFF","Wifi":{"AP":1,"SSId":"Help-Info Central Tec 2.4G","BSSId":"24:FD:0D:69:B7:2D","Channel":11,"RSSI":72,"Signal":-64,"LinkCount":1,"Downtime":"0T00:00:07"}}
00:20:06 RSL: tele/tasmota/STATE = {"Time":"1970-01-01T00:20:06","Uptime":"0T00:20:08","UptimeSec":1208,"Heap":28,"SleepMode":"Dynamic","Sleep":50,"LoadAvg":19,"MqttCount":0,"POWER1":"OFF","POWER2":"OFF","Wifi":{"AP":1,"SSId":"Help-Info Central Tec 2.4G","BSSId":"24:FD:0D:69:B7:2D","Channel":11,"RSSI":76,"Signal":-62,"LinkCount":1,"Downtime":"0T00:00:07"}}
00:25:06 RSL: tele/tasmota/STATE = {"Time":"1970-01-01T00:25:06","Uptime":"0T00:25:08","UptimeSec":1508,"Heap":28,"SleepMode":"Dynamic","Sleep":50,"LoadAvg":19,"MqttCount":0,"POWER1":"OFF","POWER2":"OFF","Wifi":{"AP":1,"SSId":"Help-Info Central Tec 2.4G","BSSId":"24:FD:0D:69:B7:2D","Channel":11,"RSSI":52,"Signal":-74,"LinkCount":1,"Downtime":"0T00:00:07"}}
00:30:06 RSL: tele/tasmota/STATE = {"Time":"1970-01-01T00:30:06","Uptime":"0T00:30:08","UptimeSec":1808,"Heap":28,"SleepMode":"Dynamic","Sleep":50,"LoadAvg":19,"MqttCount":0,"POWER1":"OFF","POWER2":"OFF","Wifi":{"AP":1,"SSId":"Help-Info Central Tec 2.4G","BSSId":"24:FD:0D:69:B7:2D","Channel":11,"RSSI":70,"Signal":-65,"LinkCount":1,"Downtime":"0T00:00:07"}}
00:35:06 RSL: tele/tasmota/STATE = {"Time":"1970-01-01T00:35:06","Uptime":"0T00:35:08","UptimeSec":2108,"Heap":28,"SleepMode":"Dynamic","Sleep":50,"LoadAvg":19,"MqttCount":0,"POWER1":"OFF","POWER2":"OFF","Wifi":{"AP":1,"SSId":"Help-Info Central Tec 2.4G","BSSId":"24:FD:0D:69:B7:2D","Channel":11,"RSSI":70,"Signal":-65,"LinkCount":1,"Downtime":"0T00:00:07"}}
00:40:06 RSL: tele/tasmota/STATE = {"Time":"1970-01-01T00:40:06","Uptime":"0T00:40:08","UptimeSec":2408,"Heap":28,"SleepMode":"Dynamic","Sleep":50,"LoadAvg":19,"MqttCount":0,"POWER1":"OFF","POWER2":"OFF","Wifi":{"AP":1,"SSId":"Help-Info Central Tec 2.4G","BSSId":"24:FD:0D:69:B7:2D","Channel":11,"RSSI":72,"Signal":-64,"LinkCount":1,"Downtime":"0T00:00:07"}}

![Tasmota](https://user-images.githubusercontent.com/19960084/162785301-1ef93b6b-607b-45fa-b926-f2785512d47d.png)

**Comment by cesarmorisco** [comment]
00:09:58 APP: Registro Serial desabilitado

**Reply by sfromis** [reply]
That is **not** the serial port being disabled, or "shutting down", only Tasmota ceasing logging messages there, unless you make it continue logging, as I already explained. 

Maybe the Portuguese translation does not have the nuances where the original english-language text is pretty clear, `Serial logging disabled`.

**Reply by ascillato** [reply]
Tasmota disables the serial LOG after 10 minutes. **_Not the serial port_**. The serial port **_continues working_**. If you send or receive any command it will work.

**Reply by ascillato2** [reply]
The Serial Port never shutdowns. The Logging report of the console shutdowns if not needed, but the port will continue working. Please, if you want to connect a microcontroller, just follow the Tasmota Oficial Documentation on that: https://tasmota.github.io/docs/TasmotaClient/

---------------

A porta serial nunca desliga. O relatório Logging do console é encerrado se não for necessário, mas a porta continuará funcionando. Por favor, se você deseja conectar um microcontrolador, basta seguir a Documentação Oficial da Tasmota sobre isso ( https://tasmota.github.io/docs/TasmotaClient/ ).

**Comment by lastmetroid** [comment]
And how do you disable the auto shutdown of the seriallog after 10m and keep it running? I dont see any word about this in the documentation: https://tasmota.github.io/docs/Commands/

If you do a:

Rule1 ON system#boot DO Backlog Baudrate 9600; SerialLog 4 endon

will the auto shutdown be disabled?

**Reply by barbudor** [reply]
What I've done in the past is a time based rule to set the log level to the expected value every 5 minutes so the countdown is restarted to 10 minutes
A rules such as 'on time#minute|5 do SerialLog 4 endon'

**Reply by sfromis** [reply]
My experience is that once after boot is enough, you do not need to keep doing it at regular intervals. The countdown is is not each time you set the log level, only once after boot.

And no reason to set the baud rate repeatedly either, as this is a persistent property, it is saved across boots if you just do it once.

(Added: Left a serial logging run for over two hours, did not stop after once doing the after-boot `SerialLog` command)
